test/cri-containerd/clone_test.go

```diff
@@ -0,0 +1,482 @@
+	requireFeatures(t, featureWCOWHypervisor)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	client := newTestRuntimeClient(t)
+	nTemplates := 2
+	var wg sync.WaitGroup
+	templatePodIDs := make([]string, nTemplates)
+	templateContainerIDs := make([]string, nTemplates)
+	clonedPodIDs := make([]string, nTemplates)
+	clonedContainerIDs := make([]string, nTemplates)
+
+	for i := 0; i < nTemplates; i++ {
+		defer cleanupPod(t, client, ctx, &templatePodIDs[i])
+		defer cleanupContainer(t, client, ctx, &templateContainerIDs[i])
+		defer cleanupPod(t, client, ctx, &clonedPodIDs[i])
+		defer cleanupContainer(t, client, ctx, &clonedContainerIDs[i])
+	}
+
+	pullRequiredImages(t, []string{imageWindowsNanoserver})
+
+	wg.Add(nTemplates)
+	for i := 0; i < nTemplates; i++ {
+		go func(index int) {
+			defer wg.Done()
+			createTemplateContainer(ctx, t, client, getTemplatePodConfig(fmt.Sprintf("templatepod-%d", index)), getTemplateContainerConfig(fmt.Sprintf("templatecontainer-%d", index)), &templatePodIDs[index], &templateContainerIDs[index])
+			verifyTemplateContainerState(ctx, t, client, templateContainerIDs[index])
+			createClonedContainer(ctx, t, client, templatePodIDs[index], templateContainerIDs[index], index, &clonedPodIDs[index], &clonedContainerIDs[index])
+		}(i)
+	}
+
+	// Wait before all template & clone creations are done.
+	wg.Wait()
+
+	for i := 0; i < nTemplates; i++ {
+		verifyContainerExec(ctx, t, client, clonedContainerIDs[i])
+	}
+}
+
+// Tries to create a clone with a different memory config than the template
+// and verifies that the request correctly fails with an error.
+func Test_VerifyCloneAndTemplateConfig(t *testing.T) {
+	requireFeatures(t, featureWCOWHypervisor)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	client := newTestRuntimeClient(t)
+	var templatePodID, templateContainerID string
+	var clonedPodID string
+
+	// send pointers so that immediate evaluation of arguments doesn't evaluate to empty strings
+	defer cleanupPod(t, client, ctx, &templatePodID)
+	defer cleanupContainer(t, client, ctx, &templateContainerID)
+	defer cleanupPod(t, client, ctx, &clonedPodID)
+
+	pullRequiredImages(t, []string{imageWindowsNanoserver})
+
+	templatePodConfig := getTemplatePodConfig("templatepod")
+
+	createTemplateContainer(ctx, t, client, templatePodConfig, getTemplateContainerConfig("templatecontainer"), &templatePodID, &templateContainerID)
+	verifyTemplateContainerState(ctx, t, client, templateContainerID)
+
+	// change pod config to make sure the request fails
+	cloneSandboxRequest := getClonedPodConfig(0, templatePodID)
+	cloneSandboxRequest.Config.Annotations["io.microsoft.virtualmachine.computetopology.memory.allowovercommit"] = "false"
+
+	_, err := client.RunPodSandbox(ctx, cloneSandboxRequest)
+	if err == nil {
+		t.Fatalf("pod cloning should fail with mismatching configurations error")
+	} else if !strings.Contains(err.Error(), "doesn't match") {
```

Comment:
Anyway (it's obviously not present in this PR) that we could export the err value so we could use it directly here. If we ever changed the error string this would break